### DISCUSSION TITLE
UPF support

### DISCUSSION THREAD

**Discussion by kareefardi** [The-OpenROAD-Project/OpenROAD#3531] [Q&A] [answered]
Is https://github.com/The-OpenROAD-Project/OpenROAD/tree/master/src/upf a work in progress (it doesn't have documentation). If it is, is it planned to have it supported?

**Comment by maliberty** [comment, marked as answer]
UPF support is a work in progress.  The reader in upf/ works but the rest of the tool needs to support it.

**Comment by maliberty** [comment]
I think the last remaining piece was pdn inserting the power switches.  @gadfort would you update on this?

**Reply by gadfort** [reply]
@maliberty inserting power switch should be working.

**Reply by maliberty** [reply]
I guess we have all the pieces in place then.  @kareefardi  If you want to be the first user that would be good.
